Fixed "My Mentors" header displaying when no mentor is assigned

MidnightChems opened the pull request:
Pull-Request for `paretOS`

## Description
- myMentors header was returned with no conditionals to check for mentors
- Added a short-circuit conditional to check if coaches name is valid before displaying text 


## Relates to
- #155

## Reviewers
- @jayeclark
- @mikhael28 

## Screenshots / other info
...I don't think I have a way to add a mentor to my account to test fully. Let me know if there's any issues and I'll take another look.

MidnightChems commented:
Like the PR says, I'm not sure how to add a mentor(or if I even can) so I can't test the code fix, so if it doesn't work let me know!

Thanks,

MidnightChems commented:
> Hi @MidnightChems I've made a generic mentee testing account that has a mentor - email me at [jay@jayeclark.dev](mailto:jay@jayeclark.dev) for the credentials! Or let me know your email address and I can assign a random mentor to that account.
> 
> There's a small issue I saw when I logged in to test - if a user _does_ have a mentor or mentors, their info is displayed but the "My Mentors" headline is not.
> 
> <img alt="Screen Shot 2022-05-12 at 12 31 00 PM" width="1182" src="https://user-images.githubusercontent.com/84106309/168124246-7f4e60b8-70c8-448f-85e4-22bde9d4e31f.png">

Hello! My email address used with the app is: wesorazem@gmail.com

Let me know when a mentor is on my account and I'll see if I can figure it out! 😀

jayeclark commented:
OK great @MidnightChems, I've added "Pareto Test Mentor" as a mentor to that account!

MidnightChems commented:
Thanks! @jayeclark ! I'll look into that asap.

mikhael28 commented on src/learn/LearnDashboard.js:
```diff
@@ -60,7 +60,7 @@ function LearnDashboard(props) {
           style={{ display: "flex", flexDirection: "column" }}
         >
-          <h2>{I18n.get("myMentors")}</h2>
+          {props.coaches.fName && <h2>{I18n.get("myMentors")}</h2>}
```
Off the top of my head, I don't think this is correct - I'm not sure you actually pulled the code down on your local machine ;) If you look at the next line of code, where we run `coaches.map` you will see that coaches is an array. So, logically, it will not have a property called `fName`. Perhaps a better way would be for you to check `props.coaches.length > 0` or something to that extent. Regardless, will you please upload a screenshot with the next revision of the PR showing that the UI is in fact displaying correctly?

MidnightChems replied:
Hello,

I see, thanks for the explanation.The code is definitely on my local machine lol. I'm new to Js so this is helping me get better. I've made some changes and will make the next revision of the PR.

MidnightChems commented:
> OK great @MidnightChems, I've added "Pareto Test Mentor" as a mentor to that account!

Hello @jayeclark 
I tried to test, but didn't see any mentors come up. Switched out of my dev branch and back to the main, and there is still no mentor info showing.

jayeclark commented:
Hmmm...  strange. I logged into the mentor account and you're showing up as a mentee for that account. 🤔  

Oh, you know what? You might have to reset some of the local storage on your browser... (basically delete the 'coaches' key & value and then reload.)  That reminds me, I need to add that to issues as a bug!!

<img width="1453" alt="Screen Shot 2022-05-14 at 5 43 07 PM" src="https://user-images.githubusercontent.com/84106309/168449114-9f96315d-2059-42c9-9048-ecefbef892ea.png">

<img width="768" alt="Screen Shot 2022-05-14 at 5 49 05 PM" src="https://user-images.githubusercontent.com/84106309/168449138-f80b6972-c6c0-4659-ad7a-74d4f81247ad.png">

MidnightChems commented:
@jayeclark @mikhael28 

Thank you both for your assistance and patience with this. I'm new to GitHub and this level of Js so this is a learning experience for me. 

I have made another commit to this branch/PR via "commit 81f6a8677c74f21feaf348a15f7ece0dd2be03f1" to fix the issue @mikhael28 brought up.

Attached is two screenshots. On is my account that has a mentor, the other is my account that has no mentor.

Let me know if there is anything else I can do.

![mentor](https://user-images.githubusercontent.com/46908663/168449556-572df7ce-dff4-42f6-b37a-436cdab44c98.png)
![nomentor](https://user-images.githubusercontent.com/46908663/168449558-cb5d3f92-2f30-463c-a60e-52a6c157cc46.png)

MidnightChems reviewed (commented):
made commit with improved code. Tested working with mentors account.

jayeclark reviewed (approved):
Looks good! (Thanks for your patience - I've been AFK with job-related stuff the past couple weeks!)
